blog/_posts/2021-08-20-Mitchell-Johnson_Jamie-Schmidt.md

```diff
@@ -0,0 +1,296 @@
+        org.GenerateOutput();
+
+        emp::DataMap dm = org.GetVar<emp::DataMap>(bits_trait);
+
+        Genome::Head head = dm.Get<Genome::Head>(bits_trait);
+
+        double fitness = 0.0;
+        double total = 0.0;
+
+        while (head.IsValid()) {
+          fitness += head.ReadBit();
+          total += 1;
+        }
+
+        // If we were supposed to count zeros, subtract ones count from total number of bits.
+        if (count_type == 0) fitness = total - fitness;
+
+        // Store the count on the organism in the fitness trait.
+        org.SetVar<double>(fitness_trait, fitness);
+
+        if (fitness > max_fitness || !max_org) {
+          max_fitness = fitness;
+          max_org = &org;
+        }
+      }
+
+      std::cout << "Max " << fitness_trait << " = " << max_fitness << std::endl;
+    }
+        
+```
+        
+The important thing here (and the thing which you will likely need to edit for any custom experiment is this bit of code: 
+        
+        
+```cpp
+        
+          Genome::Head head = dm.Get<Genome::Head>(bits_trait);
+
+          double fitness = 0.0;
+          double total = 0.0;
+
+          while (head.IsValid()) {
+            fitness += head.ReadBit();
+            total += 1;
+          }
+
+          // If we were supposed to count zeros, subtract ones count from total number of bits.
+          if (count_type == 0) fitness = total - fitness; 
+```
+        
+After we generate the output, get its DataMap, and acquire the output DataMap from it, we have the actually unique part about the Evaluator.  We grab the Head from the output DataMap (remember, we put it there with the previous example Brain!) and then we compute and set the fitness.  You can start to see how any experiment you might want to run involving evolving a vector of bits towards a particular quality is practically trivial to write, since all you'd need to do is edit the few lines of code that actually calculate the fitness.  Trying out different Genomes, Brains, and configuations thereof on that problem is then incredibly easy, since each trial is simply a tweak of the config file.
+
+## Tutorial - Creating and running a custom experiment using DynamicOrg
+
+Aagos (Auto-Adaptive Genetic Organization System), is a digital evolution experiment designed to show how high and low environmental change promote the evolution of segregated or non-segregated genes. It consists of N K long genes in a circular genome that can undergo insertion/deletion/in-place mutations, as well as gene move mutations. When a gene move mutation occurs, the gene randomly picks a new starting position in the genome. There is a maximum and minimum length the genome can grow/shrink to, and fitness is calculated by comparing the N genes to a set of gene targets that act as the environment. As a demonstration of DynamicOrg and its usefulness, we will show the steps involved in implementing Aagos using DynamicOrg.
```

Comment:
change "It consists of N K long" to "N K-long" to make this clearer (if its the case that there are N things that have length K)

Reply:
That did look and sound a bit confusing, thank you